### PR TITLE
[SDK-3858] Add documentation for modular variant of library

### LINKED ISSUE #1443: Add documentation for tree-shakable version of the library
Add documentation for the tree-shakable version of the library added in https://ably.atlassian.net/browse/SDK-3719. This includes:

- README
- generated TypeDoc documentation

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
Adds documentation (TypeDoc and in `README`) for the modular variant of the library. See commit messages for more details.

Resolves #1443.

### CODE REVIEW

File: modules.d.ts
Comment (lawrence-forooghian) [inline modules.d.ts:111]: Do you have any knowledge / thoughts that would help here, @owenpearson?

File: modules.d.ts
Comment (lawrence-forooghian) [inline modules.d.ts:124]: Do you have any knowledge / thoughts that would help here, @owenpearson?

File: modules.d.ts
Comment (owenpearson) [reply modules.d.ts:111]: `xhr_streaming` uses http streaming, ie the request is held open for a long time (60s per request i believe) and realtime messages are sent as chunks, whereas `xhr_polling` is just long polling, ie the request is held open until the server responds and then a new request is opened immediately after.

HTTP streaming is sometimes blocked by corporate firewalls so `xhr_polling` is the more likely transport to succeed but is less preferable because it more requests.

File: modules.d.ts
Comment (owenpearson) [reply modules.d.ts:124]: Yeah `fetch` is supported by the evergreen browsers, but there are still all sorts of weird web-like js platforms out there. Let's not talk about any specific scenario where someone wants to avoid `fetch` but just list it as an alternative. Also, `XHRRequest` will of course still be needed to use the web comet transports.

File: README.md
Comment (owenpearson) [inline README.md:88]: I don't like "import `ably/modules`" as a clause, it feels a bit imprecise to me? I tried to rewrite the paragraph to avoid it, but I still don't love it.

```suggestion
To use this variant of the library, import the `BaseRealtime` class from `ably/modules`, along with the additional modules you wish to use. Various modules such as `Rest`, `RealtimePresence`, etc. are available. To create a client, simply pass the additional modules to the `BaseRealtime` constructor as shown in the example below:
```

File: README.md
Comment (owenpearson) [inline README.md:86]: I do think we should emphasise here that the API is the same as for the default clients, and be explicit that you can develop an application with a default client and then switch over to the modular client only when you want to start using it in production.

Comment (owenpearson) [reply README.md:86]: I also think it's worth being clear here exactly what's required for a working realtime client, ie at least one transport and at least one http client

**Review by owenpearson — CHANGES_REQUESTED** [review]

File: README.md
Comment (lawrence-forooghian) [reply README.md]: Thanks, have updated it.

File: README.md
Comment (lawrence-forooghian) [reply README.md]: I've tweaked the wording based on your suggestion; see what you think.

File: README.md
Comment (lawrence-forooghian) [reply README.md:86]: > I do think we should emphasise here that the API is the same as for the default clients, and be explicit that you can develop an application with a default client and then switch over to the modular client only when you want to start using it in production.

Done.

> I also think it's worth being clear here exactly what's required for a working realtime client, ie at least one transport and at least one http client

I already mentioned this in the typings (and hence the TypeDoc docs), but have added it to the README too.

File: modules.d.ts
Comment (lawrence-forooghian) [reply modules.d.ts]: OK, I've left the wording as it is.

File: modules.d.ts
Comment (lawrence-forooghian) [reply modules.d.ts]: Thanks for the information. I've added some wording; see if you think it's enough?

File: README.md
Comment (owenpearson) [inline README.md:109]: hard to explain why i think this is better, but:
```suggestion
You must provide:

- at least one HTTP request implementation; that is, one of `FetchRequest` or `XHRRequest`;
- at least one realtime transport implementation; that is, one of `WebSocketTransport`, `XHRStreaming`, or `XHRPolling`.
```

File: modules.d.ts
Comment (owenpearson) [reply modules.d.ts]: That looks great, thanks 👍

**Review by owenpearson — APPROVED** [review]
Looks excellent, nice one 👍 I've got one additional suggested tweak but feel free to ignore if you disagree. I'll approve now so you can merge when you've resolved the eslint ignore comment

File: README.md
Comment (lawrence-forooghian) [reply README.md:109]: Yeah, I agree it reads better — will change.

**Pull request merged**
